components/Alert/index.d.ts

```diff
@@ -3,7 +3,7 @@ import { IconNames } from '../Icon';
 
 export interface AlertProps {
     children: ReactNode;
-    onClose?: () => {} | MouseEventHandler<HTMLButtonElement> | TouchEventHandler<HTMLButtonElement>;
+    onClose?: MouseEventHandler<HTMLButtonElement> | TouchEventHandler<HTMLButtonElement>;
```

Comment:
I think is better keep generic function here. How about: 
```suggestion
    onClose?: function | MouseEventHandler<HTMLButtonElement> | TouchEventHandler<HTMLButtonElement>;
```